x-pack/platform/plugins/shared/onechat/server/utils/kibana_mcp_http_transport.ts

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Transport } from '@modelcontextprotocol/sdk/shared/transport.js';
+import type { JSONRPCMessage, RequestId } from '@modelcontextprotocol/sdk/types.js';
+import type { KibanaResponseFactory, KibanaRequest } from '@kbn/core-http-server';
+import {
+  isJSONRPCRequest,
+  isJSONRPCResponse,
+  isJSONRPCError,
+  JSONRPCMessageSchema,
+  isInitializeRequest,
+} from '@modelcontextprotocol/sdk/types.js';
+import { Logger } from '@kbn/logging';
+import { AuthInfo } from '@modelcontextprotocol/sdk/server/auth/types';
+import { randomUUID } from 'node:crypto';
+import type { IKibanaResponse } from '@kbn/core/server';
+
+// SDK error codes
+export const CONNECTION_CLOSED = -32000;
+
+// Standard JSON-RPC error codes
+export const PARSE_ERROR = -32700;
+export const INVALID_REQUEST = -32600;
+export const METHOD_NOT_FOUND = -32601;
+export const INVALID_PARAMS = -32602;
+export const INTERNAL_ERROR = -32603;
```

Comment:
NIT: Are those exposed from the SDK? If so, we should probably import them from there instead of redefining them, no?

Reply:
Ah right, they in fact [are](https://github.com/modelcontextprotocol/typescript-sdk/blob/1878143f1a8ffc3ce5f7acd4dc61cef67b589b4b/src/types.ts#L117)! Will update